subsys/net/ip/icmp.c

```diff
@@ -0,0 +1,596 @@
+/** @file
+ * @brief ICMP related functions
+ */
+
+/*
+ * Copyright (c) 2023 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/* Use highest log level if both IPv4 and IPv6 are defined */
+#if defined(CONFIG_NET_IPV4)
+#define MY_LOG_LEVEL CONFIG_NET_ICMPV4_LOG_LEVEL
+#endif
+
+#if defined(CONFIG_NET_IPV6)
+#if defined(MY_LOG_LEVEL)
+#if MY_LOG_LEVEL < CONFIG_NET_ICMPV6_LOG_LEVEL
+#undef MY_LOG_LEVEL
+#define MY_LOG_LEVEL CONFIG_NET_ICMPV6_LOG_LEVEL
+#endif
+#else
+#define MY_LOG_LEVEL CONFIG_NET_ICMPV6_LOG_LEVEL
+#endif
+#endif
+
+#if !defined(MY_LOG_LEVEL)
+#define MY_LOG_LEVEL LOG_LEVEL_INF
+#endif
+
+#include <zephyr/logging/log.h>
+LOG_MODULE_REGISTER(net_icmp, MY_LOG_LEVEL);
+
+#include <errno.h>
+#include <zephyr/random/rand32.h>
+#include <zephyr/sys/slist.h>
+#include <zephyr/net/net_pkt.h>
+#include <zephyr/net/icmp.h>
+
+#include "net_private.h"
+#include "icmpv6.h"
+#include "icmpv4.h"
+#include "ipv4.h"
+#include "ipv6.h"
+#include "net_stats.h"
+
+static K_MUTEX_DEFINE(lock);
+static sys_slist_t handlers;
+
+#if defined(CONFIG_NET_OFFLOADING_SUPPORT)
+static sys_slist_t offload_handlers;
```

Comment:
There is a macro for static initialization of a single-linked list.

```suggestion
static sys_slist_t handlers = SYS_SLIST_STATIC_INIT(&handlers);

#if defined(CONFIG_NET_OFFLOADING_SUPPORT)
static sys_slist_t offload_handlers = SYS_SLIST_STATIC_INIT(&offload_handlers);
```

Reply:
Good point, will fix.